Source/GSAutoLayoutVFLParser.m

```diff
@@ -0,0 +1,755 @@
+
+  if (![_scanner scanString: @")" intoString: nil])
+    {
+      [self failParseWithMessage: @"A predicate on a view must end with ')'"];
+    }
+
+  return viewPredicateConstraints;
+}
+
+- (NSLayoutConstraint *) createConstraintFromParsedPredicate:
+    (GSObjectOfPredicate *)predicate
+{
+  NSLayoutConstraint *constraint = nil;
+  NSLayoutAttribute attribute = _isVerticalOrientation
+                                    ? NSLayoutAttributeHeight
+                                    : NSLayoutAttributeWidth;
+  if (predicate->view != nil)
+    {
+      constraint = [NSLayoutConstraint constraintWithItem: _view
+                                                attribute: attribute
+                                                relatedBy: predicate->relation
+                                                   toItem: predicate->view
+                                                attribute: attribute
+                                               multiplier: 1.0
+                                                 constant: predicate->constant];
+    }
+  else
+    {
+      constraint = [NSLayoutConstraint
+          constraintWithItem: _view
+                   attribute: attribute
+                   relatedBy: predicate->relation
+                      toItem: nil
+                   attribute: NSLayoutAttributeNotAnAttribute
+                  multiplier: 1.0
+                    constant: predicate->constant];
+    }
+
+  if (predicate->priority)
+    {
+      constraint.priority = [predicate->priority doubleValue];
+    }
+
+  return constraint;
+}
+
+- (GSObjectOfPredicate *) parseObjectOfPredicate
+{
+  NSLayoutRelation relation = [self parseRelation];
+
+  CGFloat parsedConstant;
+  NSView *predicatedView = nil;
+  BOOL scanConstantResult = [_scanner scanDouble: &parsedConstant];
+  if (!scanConstantResult)
+    {
+      NSString *identiferName = [self parseIdentifier];
+      if (![self isValidIdentifier: identiferName])
+        {
+          [self failParseWithMessage:
+                    @"Invalid metric or view identifier. Metric/View "
+                    @"identifiers must be a valid C identifier and may only "
+                    @"contain letters, numbers and underscores"];
+        }
+
+      NSNumber *metric = [_metrics objectForKey: identiferName];
+      if (metric != nil)
+        {
+          parsedConstant = [metric doubleValue];
+        }
+      else if ([_views objectForKey: identiferName])
+        {
+          parsedConstant = 0;
+          predicatedView = [_views objectForKey: identiferName];
+        }
+      else
+        {
+          NSString *message = [NSString
+              stringWithFormat:
+                  @"Failed to find constant or metric for identifier '%@'",
+                  identiferName];
+          [self failParseWithMessage: message];
+        }
+    }
+
+  NSNumber *priorityValue = [self parsePriority];
+
+  GSObjectOfPredicate *predicate = calloc (1, sizeof (GSObjectOfPredicate));
```

Comment:
The `calloc` here is not strictly necessary. You could create this structure on the stack in `parsePredicateList` and pass it down. That way also the `free` won't be needed.
You don't need to change this, this is just an optimisation hint.